File: php/features.php
<?php
declare(strict_types=1);

// FreeElevation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class FreeElevationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new FreeElevationBaseFeature();
            case "test":
                return new FreeElevationTestFeature();
            default:
                return new FreeElevationBaseFeature();
        }
    }
}

File: py/features.py
# FreeElevation SDK feature factory

from feature.base_feature import FreeElevationBaseFeature
from feature.test_feature import FreeElevationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FreeElevationBaseFeature(),
        "test": lambda: FreeElevationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
